core/dbt/config/profile.py

```diff
@@ -273,6 +278,11 @@ def render_profile(
         elif "target" in raw_profile:
             # render the target if it was parsed from yaml
             target_name = renderer.render_value(raw_profile["target"])
+        elif is_secondary and len(raw_profile["outputs"]) == 1:
```

Comment:
Is the "output" key guaranteed to exist? Let's make a .get("outputs", []) call here to be safe.

Reply:
Separately -- is this case mutually exclusive from the above? Or does this is_secondary handling still need to happen even if "target" is in raw_profile?

Reply:
> Is the "output" key guaranteed to exist?
I think so, an error gets thrown earlier – `test_no_outputs` and `test_secondary_profiles_no_outputs` are checking for this scenario.

> Let's make a .get("outputs", []) call here to be safe.
Agreed, I'll do this regardless.

Reply:
> Separately -- is this case mutually exclusive from the above?

I believe so, yes. If user specifies `target: foo` but `outputs` only has `target: bar`, we should raise an error instead of defaulting to `bar`.

> Or does this is_secondary handling still need to happen even if "target" is in raw_profile?

I'm not sure I understand the question – if `"target" in raw_profile == True`, then `is_secondary and len(raw_profile["outputs"]) == 1` is never evaluated?

Reply:
> if "target" in raw_profile == True, then is_secondary and len(raw_profile["outputs"]) == 1 is never evaluated

That's the current behaviour as-implemented. Just confirming that it's intentional. Sounds like it is

Reply:
Yup, it's intentional. Thanks for checking!